File: 03.Hashing/app_compara.cpp
#include <iostream>
#include <fstream>
#include <algorithm>
#include "separatechaining_hashtable.h"
#include "linearprobing_hashtable.h"

using namespace std;

int main()
{
    SeparateChainingHashTable<string, int> ht;
    //LinearProbingHashTable<string, int> ht;
    ifstream arq("DomCasmurro_utf8.txt");
    if(!arq.is_open()) {
        cout << "Erro abrindo arquivo!" << endl;
        return 1;
    }

    string pal;

    while(arq >> pal) {
        // 1. Remove pontuação
        pal.erase(remove_if(pal.begin(), pal.end(), [](unsigned char c) {
            return ispunct(c);
        }), pal.end());

        // 2. Remove espaços
        pal.erase(remove_if(pal.begin(), pal.end(), [](unsigned char ch) {
            return isspace(ch);
        }), pal.end());

        // 3. Converte para minúsculas
        transform(pal.begin(), pal.end(), pal.begin(), [](unsigned char c) {
            return tolower(c);
        });
        cout << pal << endl;
    }

}

File: 03.Hashing/appcompara.py
from separatechaining import SeparateChainingHashTable
from linearprobing import LinearProbingHashTable
import re

if __name__=="__main__":

    ht = SeparateChainingHashTable()

    with open("DomCasmurro_utf8.txt") as arq:
        for linha in arq:
            pals = linha.split()
            for pal in pals:
                pal = pal.lower().strip()
                # Remove tudo que não for letra, número, _,
                # e espaço em branco
                pal = re.sub(r"[^\w]", "", pal)
                if pal == "":
                    continue
                print(pal)